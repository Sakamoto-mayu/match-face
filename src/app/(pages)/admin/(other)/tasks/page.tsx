const TasksPage = () => {
  return <div>タスク一覧画面</div>;
};

export default TasksPage;
