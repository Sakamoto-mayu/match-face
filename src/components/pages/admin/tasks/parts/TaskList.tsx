const TaskList = () => {
    return (
        <div>タスク一覧</div>
    );
};

export default TaskList
