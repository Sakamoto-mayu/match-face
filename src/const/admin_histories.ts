export const skills = [
    {id:1,skill:"JavaScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,skill:"TypeScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,skill:"React",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,skill:"Next.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,skill:"Vue.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,skill:"Angular",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:7,skill:"Nuxt.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:8,skill:"jQuery",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,skill:"Redux",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:10,skill:"vuex",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:11,skill:"MaterialUI",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:12,skill:"tailwind",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:13,skill:"Bootstrap",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:14,skill:"Vercel",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const departments = [
    {id:1,name:"FR",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:2,name:"Java",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:3,name:"PHP",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:4,name:"CL",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:5,name:"ML",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:6,name:"QA",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
]

export const projects = [
    {id:1,name:"案件A",detail:"案件Aの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:"案件B",detail:"案件Bの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:"案件C",detail:"案件Cの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:"案件D",detail:"案件Dの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:"案件E",detail:"案件Eの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
]

export const project_skills = [
    {id:1,project_id:1,skill_id:[1,2,3],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,project_id:2,skill_id:[4,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,project_id:3,skill_id:[8,13],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,project_id:4,skill_id:[1,2,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,project_id:5,skill_id:[2,3,4,6],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answer_requests = [
    {id:1,user_id:[1,2],administrator_id:5,project_id:1,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,user_id:[1,2],administrator_id:5,project_id:2,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,user_id:[3],administrator_id:5,project_id:3,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,user_id:[1,2,4],administrator_id:5,project_id:4,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,user_id:[4],administrator_id:5,project_id:5,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-08-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answers = [
    {id:1,context:'回答が入ります1',question_id:1,answer_request_id:1,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,context:'回答が入ります2',question_id:1,answer_request_id:1,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:3,context:'回答が入ります3',question_id:2,answer_request_id:2,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-13T16:52:46.053Z'},
    {id:4,context:'回答が入ります4',question_id:2,answer_request_id:2,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,context:'回答が入ります5',question_id:3,answer_request_id:3,user_id:3,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,context:'回答が入ります6',question_id:4,answer_request_id:4,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:7,context:'回答が入ります7',question_id:4,answer_request_id:4,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:8,context:'回答が入ります8',question_id:4,answer_request_id:4,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,context:'回答が入ります9',question_id:5,answer_request_id:5,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
]

export const users = [
    {id:1,name:'田中太郎',password:'tanaka',email:'tanaka@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:'山田次郎',password:'yamada',email:'yamada@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:'佐藤三郎',password:'sato',email:'sato@example.com',hire_date:'2023-01-01',department_id:2,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:'鈴木四郎',password:'suzuki',email:'suzuki@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:'高橋五郎',password:'takahashi',email:'takahashi@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]
