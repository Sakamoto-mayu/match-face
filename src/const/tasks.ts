export const departments = [
  {
    id: 1,
    name: "Java",
  },
  {
    id: 2,
    name: "PHP",
  },
  {
    id: 3,
    name: "FR",
  },
  {
    id: 4,
    name: "CL",
  },
  {
    id: 5,
    name: "ML",
  },
  {
    id: 6,
    name: "QA",
  },
];

export const tasks = [
  {
    id: 1,
    project_name: "hoge",
    department: "Java",
    question: ["hoge", "hoge", "hoge"],
    users: ["佐藤太郎", "鈴木次郎", "伊藤三郎"],
    answer_deadline: "2100-01-01",
  },
  {
    id: 2,
    project_name: "foo",
    department: "PHP",
    question: ["foo", "foo", "foo"],
    users: ["佐藤太郎", "鈴木次郎", "伊藤三郎", "田中四郎"],
    answer_deadline: "2100-01-01",
  },
  {
    id: 3,
    project_name: "bar",
    department: "FR",
    question: ["bar", "bar", "bar"],
    users: ["佐藤太郎", "鈴木次郎", "伊藤三郎"],
    answer_deadline: "2100-01-01",
  },
  {
    id: 4,
    project_name: "hogehoge",
    department: "CL",
    question: ["hogehoge", "hogehoge", "hogehoge"],
    users: ["佐藤太郎", "鈴木次郎", "伊藤三郎"],
    answer_deadline: "2200-01-01",
  },
  {
    id: 5,
    project_name: "foofoo",
    department: "ML",
    question: ["foofoo", "foofoo", "foofoo"],
    users: ["佐藤太郎", "鈴木次郎", "伊藤三郎", "田中四郎"],
    answer_deadline: "2200-01-01",
  },
  {
    id: 6,
    project_name: "barbar",
    department: "QA",
    question: ["barbar", "barbar", "barbar"],
    users: ["佐藤太郎", "鈴木次郎", "伊藤三郎"],
    answer_deadline: "2200-01-01",
  },
];
