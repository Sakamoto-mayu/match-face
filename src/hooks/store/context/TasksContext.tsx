"use client";

import React from "react";

// 職種フィルター用context
const JobsFilterContext = React.createContext<
  [string[], React.Dispatch<React.SetStateAction<string[]>>] | undefined
>(undefined);

export const JobsFilterProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [filterlist, setFilterList] = React.useState<string[]>([]);
  return (
    <JobsFilterContext.Provider value={[filterlist, setFilterList]}>
      {children}
    </JobsFilterContext.Provider>
  );
};

export const useJobsFilter = () => {
  const context = React.useContext(JobsFilterContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a Provider");
  }
  return context;
};
// ここまで
