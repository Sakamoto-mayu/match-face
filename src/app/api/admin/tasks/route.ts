import { Answer_RequestsType } from "@/types/admin/tasks/types";
import { NextRequest, NextResponse } from "next/server";

/**
 * @author Hayato Kobayashi
 * @param req Next API request
 * @returns Next API responce
 *
 * @todo_1 検索ワード(searchKeyword), 所属(departmentId)で検索条件を指定
 */
export async function GET(req: NextRequest) {
  // クエリから検索条件を取得
  const searchParams = req.nextUrl.searchParams;
  const searchKeyword = searchParams.get("searchKeyword");
  const departmentId = searchParams.get("departmentId");
  /**
   * searchKeywordから加工されたクエリ
   * ex) "a_b_c" → "searchKeyword=a&searchKeyword=b&searchKeyword=c"
   */
  const searchKeywordQuery = searchKeyword
    ? searchKeyword
        .split("_")
        .map((id, i) => {
          if (i === 0) return `searchKeyword=${id}`;
          return `&searchKeyword=${id}`;
        })
        .join("")
    : "";
  /**
   * departmentIdから加工されたクエリ
   * ex) "1_2_3" → "searchKeyword=1&searchKeyword=2&searchKeyword=3"
   * @note searchKeywordQueryがある場合は先頭に"&"がつく。
   */
  const departmentIdQuery = departmentId
    ? departmentId
        .split("_")
        .map((id, i) => {
          if (i === 0) return `${searchKeywordQuery && "&"}departmentId=${id}`;
          return `&departmentId=${id}`;
        })
        .join("")
    : "";

  const query = searchKeywordQuery + departmentIdQuery;

  const res = await fetch(
    `${process.env.BE_URL}/answer_requests${query && "?" + query}`,
    {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  // console.log("res", res.status);
  const answerRequests: Answer_RequestsType = await res.json();
  const tasks = answerRequests.answerRequests; // fetch時に"answerRequests.answerRequestsの形になるので修正"
  return NextResponse.json(tasks);
}
