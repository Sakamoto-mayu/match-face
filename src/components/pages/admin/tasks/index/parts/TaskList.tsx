"use client";
import { useJobsFilter } from "@/hooks/store/context/TasksContext";
import { Task } from "@/types/admin/tasks/register/types";
import { useEffect, useState } from "react";

// Todo: タスクリストを非同期通信で取得
const TaskList = ({ tasks }: { tasks: Task[] }) => {
  const [jobsFilter] = useJobsFilter();
  
  // useEffect(() => {
  //   console.log("jobsFilter", jobsFilter);
  //   console.log("length", jobsFilter.length);
  // }, [jobsFilter]);

  const tasksFilteredByJob = tasks.filter((task) => {
    if (jobsFilter.length === 0) return task;
    return jobsFilter.includes(task.department);
  });

  return (
    <div>
      <table className="border-collapse border-2 border-deep-gray text-sm">
        <thead className="bg-light-gray">
          <tr>
            <th className="border-2 border-deep-gray p-2 text-center w-72">
              案件名
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-16">
              質問数
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-16">
              職種
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-72">
              回答ユーザー
            </th>
            <th className="border-2 border-deep-gray p-2 text-center w-28">
              期日
            </th>
          </tr>
        </thead>
        <tbody>
          {tasksFilteredByJob.map((task) => (
            <tr key={task.id}>
              <td className="border-2 border-deep-gray p-2 text-center">
                {task.project_name}
              </td>
              <td className="border-2 border-deep-gray p-2 text-center">
                {task.question.length}
              </td>
              <td className="border-2 border-deep-gray p-2 text-center">
                {task.department}
              </td>
              <td className="border-2 border-deep-gray p-2 text-center">
                {/* 回答ユーザーは3名まで表示。それ以降は"..."で省略 */}
                {task.users.length > 3
                  ? task.users.slice(0, 3).join("、") + "..."
                  : task.users.join("、")}
              </td>
              <td className="border-2 border-deep-gray p-2 text-center">
                {task.answer_deadline}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default TaskList;
