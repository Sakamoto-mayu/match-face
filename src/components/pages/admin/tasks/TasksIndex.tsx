import OrangeButton from "@/components/ui/button/OrangeButton";
import SearchByJobs from "./parts/SearchByJobs";
import TaskList from "./parts/TaskList";

const TasksIndex = () => {
  return (
    <main>
      <div className="mb-8">
        <SearchByJobs />
      </div>
      <div className="flex justify-center mb-8">
        <OrangeButton label="新規追加" />
      </div>
      <div className="flex justify-center">
        <TaskList />
      </div>
    </main>
  );
};

export default TasksIndex;
