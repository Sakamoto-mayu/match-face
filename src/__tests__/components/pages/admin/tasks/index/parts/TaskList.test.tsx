import TaskList from "@/components/pages/admin/tasks/index/parts/TaskList";
import { cleanup, render, screen } from "@testing-library/react";
import { tasks } from "@/const/tasks";
import { FilterProvider, FilterType } from "@/hooks/store/context/TasksContext";
import { useFilter } from "@/hooks/store/context/TasksContext";
import "@testing-library/jest-dom";

describe("TaskList.tsx", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it("レンダリング時", () => {
    const { container } = render(
      <FilterProvider>
        <TaskList tasks={tasks} />
      </FilterProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
