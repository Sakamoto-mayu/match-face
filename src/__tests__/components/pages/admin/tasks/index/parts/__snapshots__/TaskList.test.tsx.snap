// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList.tsx レンダリング時 1`] = `
<div>
  <div>
    <table
      class="border-collapse border-2 border-deep-gray text-sm"
    >
      <thead
        class="bg-light-gray"
      >
        <tr>
          <th
            class="border-2 border-deep-gray p-2 text-center w-72"
          >
            案件名
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-16"
          >
            質問数
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-16"
          >
            職種
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-72"
          >
            回答ユーザー
          </th>
          <th
            class="border-2 border-deep-gray p-2 text-center w-28"
          >
            期日
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            hoge
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            3
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            Java
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            佐藤太郎、鈴木次郎、伊藤三郎
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2100-01-01
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            foo
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            3
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            PHP
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            佐藤太郎、鈴木次郎、伊藤三郎...
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2100-01-01
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            bar
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            3
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            FR
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            佐藤太郎、鈴木次郎、伊藤三郎
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2100-01-01
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            hogehoge
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            3
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            CL
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            佐藤太郎、鈴木次郎、伊藤三郎
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2100-01-01
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            foofoo
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            3
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            ML
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            佐藤太郎、鈴木次郎、伊藤三郎...
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2100-01-01
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            barbar
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            3
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            QA
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            佐藤太郎、鈴木次郎、伊藤三郎
          </td>
          <td
            class="border-2 border-deep-gray p-2 text-center"
          >
            2100-01-01
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
